### PR TITLE
2.x.x auth-jwt

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

File: auth-jwt/Sources/App/Middleware/BasicAuthenticator.swift [hunk 1]
```diff
@@ -15,20 +15,24 @@
...
+        let user = try await User.query(on: self.fluent.db())
             .filter(\.$name == basic.username)
             .first()
-        // can only verify password if user has password associated with it
-        guard let passwordHash = user?.passwordHash else { return nil }
-        guard Bcrypt.verify(basic.password, hash: passwordHash) else { return nil }
-        return user
+        guard let user = user, let passwordHash = user.passwordHash else { return nil }
+        // Do Bcrypt verify on a separate thread to not block the general task executor 
+        guard try await NIOThreadPool.singleton.runIfActive({ Bcrypt.verify(basic.password, hash: passwordHash) }) else { return nil }
```
Comment (Joannis) [inline auth-jwt/Sources/App/Middleware/BasicAuthenticator.swift:35]: I don't think we need to lean on a NIOThreadPool with async-await

File: auth-jwt/Sources/App/Application+build.swift [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+struct HTTPClientService: Service {
+    let client: HTTPClient
+
+    func run() async throws {
+        await GracefulShutdownWaiter().wait()
+        try await client.shutdown()
+    }
+}
+/*extension HBApplication {
```
Comment (Joannis) [inline auth-jwt/Sources/App/Application+build.swift:98]: Commented code

File: auth-jwt/Sources/App/Middleware/BasicAuthenticator.swift (hunk 1 above)
Comment (adam-fowler) [reply auth-jwt/Sources/App/Middleware/BasicAuthenticator.swift:35]: Yes we do, we shouldn't be blocking the global task executor.

File: auth-jwt/Sources/App/Application+build.swift (hunk 2 above)
Comment (adam-fowler) [reply auth-jwt/Sources/App/Application+build.swift]: Removed

**Review by Joannis — APPROVED** [review]

**Pull request merged**